x-pack/plugins/observability_shared/public/components/page_template/page_template.tsx

```diff
@@ -194,38 +198,41 @@ export function ObservabilityPageTemplate({
       >
         {({ isTourVisible }) => {
           return (
-            <KibanaPageTemplate
-              restrictWidth={false}
-              {...pageTemplateProps}
-              solutionNav={
-                showSolutionNav
-                  ? {
-                      icon: 'logoObservability',
-                      items: sideNavItems,
-                      name: sideNavTitle,
-                      // Only false if tour is active
-                      canBeCollapsed: isTourVisible === false,
-                    }
-                  : undefined
-              }
-            >
-              <KibanaErrorBoundaryProvider analytics={services.analytics}>
-                <KibanaErrorBoundary>
-                  <KibanaPageTemplate.Section
-                    component="div"
-                    alignment={pageTemplateProps.isEmptyState ? 'center' : 'top'}
-                    {...pageSectionProps}
-                  >
-                    {children}
-                  </KibanaPageTemplate.Section>
-                </KibanaErrorBoundary>
-              </KibanaErrorBoundaryProvider>
-              {bottomBar && (
-                <KibanaPageTemplate.BottomBar {...bottomBarProps}>
-                  {bottomBar}
-                </KibanaPageTemplate.BottomBar>
-              )}
-            </KibanaPageTemplate>
+            <div ref={containerRef}>
+              <KibanaPageTemplate
+                restrictWidth={false}
+                {...pageTemplateProps}
+                solutionNav={
+                  showSolutionNav
+                    ? {
+                        icon: 'logoObservability',
+                        items: sideNavItems,
+                        name: sideNavTitle,
+                        // Only false if tour is active
+                        canBeCollapsed: isTourVisible === false,
+                      }
+                    : undefined
+                }
+              >
+                <KibanaErrorBoundaryProvider analytics={services.analytics}>
+                  <KibanaErrorBoundary>
+                    <KibanaPageTemplate.Section
+                      component="div"
+                      alignment={pageTemplateProps.isEmptyState ? 'center' : 'top'}
+                      {...pageSectionProps}
+                    >
+                      <SearchBarPortal containerRef={containerRef}>{topSearchBar}</SearchBarPortal>
```

Comment:
Isn't it better to not render the `SearchBarPortal` if **topSearchBar** is undefined?

Reply:
i added the condition in component itself to return null 
i find that bit more cleaner